sdk/synapse/synapse-spark/test/utils/testAuthentication.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { ClientSecretCredential } from "@azure/identity";
+import { SparkClient } from "../../src";
+import { env, record, RecorderEnvironmentSetup } from "@azure/test-utils-recorder";
+import { uniqueString } from "./recorderUtils";
+import { getWorkspaceName, getSparkpoolName } from "./utils.common";
+
+export async function authenticate(that: any): Promise<any> {
+  const secretSuffix = uniqueString();
+  const recorderEnvSetup: RecorderEnvironmentSetup = {
+    replaceableVariables: {
+      AZURE_CLIENT_ID: "azure_client_id",
+      AZURE_CLIENT_SECRET: "azure_client_secret",
+      AZURE_TENANT_ID: "azure_tenant_id",
+      WORKSPACE_NAME: "workspace_name",
+      SPARKPOOL_NAME: "sparkpool_name"
+    },
+    customizationsOnRecordings: [
+      (recording: any): any =>
+        recording.replace(/"access_token":"[^"]*"/g, `"access_token":"access_token"`),
+      (recording: any): any =>
+        secretSuffix === "" ? recording : recording.replace(new RegExp(secretSuffix, "g"), "")
+    ],
+    queryParametersToSkip: []
+  };
+  const recorder = record(that, recorderEnvSetup);
+  const credential = await new ClientSecretCredential(
+    env.AZURE_TENANT_ID,
+    env.AZURE_CLIENT_ID,
+    env.AZURE_CLIENT_SECRET
+  );
+
+  const worksapceName = getWorkspaceName();
```

Comment:
typo: workspaceName

Reply:
Thanks for catching it!

---
In reply to: [521615392](https://github.com/Azure/azure-sdk-for-js/pull/12487#discussion_r521615392) [](ancestors = 521615392)